backend/canisters/identity/impl/src/updates/remove_identity_link.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{mutate_state, RuntimeState};
+use canister_tracing_macros::trace;
+use ic_cdk::update;
+use identity_canister::remove_identity_link::{Args, Response};
+
+#[update]
+#[trace]
+fn remove_identity_link(args: Args) -> Response {
+    mutate_state(|state| remove_identity_link_impl(args, state))
+}
+
+fn remove_identity_link_impl(args: Args, state: &mut RuntimeState) -> Response {
+    let caller = state.env.caller();
+    let auth_principal = state.data.user_principals.get_auth_principal(&caller);
+    let user_principal = state
+        .data
+        .user_principals
+        .get_by_auth_principal(&caller)
+        .filter(|u| u.user_id.is_some());
+
+    let principals = auth_principal.zip(user_principal);
+
+    if let Some((auth, user)) = principals {
+        if user.principal == args.linked_principal {
```

Comment:
Each user has a single "user principal" but can have multiple "authentication principals".
The call into the Identity canister will be made using whichever auth principal they signed in with.
So we should return `CannotUnlinkActivePrincipal` if `caller == args.linked_principal`.